x-pack/plugins/rollup/__jest__/integration_tests/job_create.test_helpers.js

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import axios from 'axios';
+
+import { registerTestBed } from '../utils';
+import { rollupJobsStore } from '../../public/crud_app/store';
+import {
+  setHttp,
+} from '../../public/crud_app/services';
+
+import { JobCreate } from '../../public/crud_app/sections';
+
+// axios has a $http like interface so using it to simulate $http
+setHttp(axios.create());
+
+// This is the Rollup job we will be creating in our tests
+const JOB_TO_CREATE = {
+  id: 'test-job',
+  indexPattern: 'test-pattern-*',
+  rollupIndex: 'rollup-index',
+  interval: '24h'
+};
+
+const initUserActions = (component, findTestSubject) => {
+  const clickNextStep = () => {
+    const button = findTestSubject('rollupJobNextButton');
+    button.simulate('click');
+    component.update();
+  };
+
+  const clickPreviousStep = () => {
+    const button = findTestSubject('rollupJobBackButton');
+    button.simulate('click');
+    component.update();
+  };
+
+  const clickSave = () => {
+    const button = findTestSubject('rollupJobSaveButton');
+    button.simulate('click');
+    component.update();
+  };
+
+  return {
+    clickNextStep,
+    clickPreviousStep,
+    clickSave,
+  };
+};
+
+const initFillFormFields = form => async (step) => {
+  switch (step) {
+    case 'logistics':
+      form.setInputValue('rollupJobName', JOB_TO_CREATE.id);
+      await form.setInputValue('rollupIndexPattern', JOB_TO_CREATE.indexPattern, true);
+      form.setInputValue('rollupIndexName', JOB_TO_CREATE.rollupIndex);
+      break;
+    case 'date-histogram':
+      form.setInputValue('rollupJobInterval', JOB_TO_CREATE.interval);
+      break;
+    default:
+      return;
+  }
+};
+
+const initGoToStep = (fillFormFields, clickNextStep) => async (step) => {
+  if (!step || step === 1) {
+    return;
+  }
+
+  await fillFormFields('logistics');
+  clickNextStep();
+
+  if (step > 2) {
+    await fillFormFields('date-histogram');
+    clickNextStep();
+  }
+
+  if (step > 3) {
+    clickNextStep();
+  }
+
+  if (step > 4) {
+    clickNextStep();
+  }
+
+  if (step > 5) {
+    clickNextStep();
+  }
+};
```

Comment:
Did the loop idea not work out?

Reply:
Forgot to update that part. Just did it 😊